cmdtests/gitbase_back_comp_test.go

```diff
@@ -0,0 +1,274 @@
+// +build regression
+
+package cmdtest
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	cmdtest "github.com/src-d/engine/cmdtests"
+
+	"gopkg.in/src-d/regression-core.v0"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/stretchr/testify/suite"
+)
+
+var prevVersion = "latest"
+var currVersion = "local:HEAD"
+
+func init() {
+	if os.Getenv("PREV_ENGINE_VERSION") != "" {
+		prevVersion = os.Getenv("PREV_ENGINE_VERSION")
+	}
+
+	if os.Getenv("CURR_ENGINE_VERSION") != "" {
+		currVersion = os.Getenv("CURR_ENGINE_VERSION")
+	}
+}
+
+type GitbaseBackCompTestSuite struct {
+	cmdtest.RegressionSuite
+	timeout     time.Duration
+	testDir     string
+	testDataDir string
+}
+
+func TestGitbaseBackCompTestSuite(t *testing.T) {
+	s := GitbaseBackCompTestSuite{timeout: 10 * time.Minute}
+	suite.Run(t, &s)
+}
+
+func (s *GitbaseBackCompTestSuite) createRepo(name string) {
+	s.T().Helper()
+	require := s.Require()
+
+	repoPath := filepath.Join(s.testDir, name)
+	err := os.Mkdir(repoPath, os.ModePerm)
+	require.NoError(err)
+
+	cmd := exec.Command("git", "init", repoPath)
+	err = cmd.Run()
+	require.NoError(err)
+}
+
+func (s *GitbaseBackCompTestSuite) SetupSuite() {
+	token := os.Getenv("GITHUB_TOKEN")
+
+	// This is required because regression-core assumes that the working
+	// directory is the root of the project
+	s.setupWd()
+
+	config := regression.NewConfig()
+	cachePath := "regression-testing-cache"
+	config.BinaryCache = filepath.Join(cachePath, "binary")
+	config.RepositoriesCache = filepath.Join(cachePath, "repos")
+
+	tool := regression.Tool{
+		Name:        "engine",
+		BinaryName:  "srcd",
+		GitURL:      "https://github.com/src-d/engine",
+		ProjectPath: "github.com/src-d/engine",
+		BuildSteps: []regression.BuildStep{
+			{
+				Dir:     "",
+				Command: "make",
+				Args:    []string{"build", "docker-build"},
+			},
+		},
+	}
+
+	releases := regression.NewReleases("src-d", "engine", token)
+
+	s.PrevCmd = cmdtest.NewCommander(s.getBinary(config, tool, releases, prevVersion))
+	s.CurrCmd = cmdtest.NewCommander(s.getBinary(config, tool, releases, currVersion))
+}
+
+func (s *GitbaseBackCompTestSuite) setupWd() {
+	s.T().Helper()
+
+	_, filename, _, _ := runtime.Caller(0)
```

Comment:
why do we need `runtime.Caller(0)` here? Isn't it equal to `os.Getwd`? If not could you please add a comment. It's not obvious at all.

Reply:
you're right, it took a couple of seconds even for me to remember the reason...

Reply:
addressed [here](https://github.com/src-d/engine/pull/363/commits/864cd8a628be86ce8986bb50edae6fb52f48feca).

Reply:
I still don't get it. We run both integration&regression tests using `go test`
So why do you need `runtime.Caller`?
It still looks like:
```
_, filename, _, _ := runtime.Caller(0)
path.Dir(filename)
```
returns the same results as `os.Getwd`

And this new comment should belong to `setupWd` function, not `runtime.Caller`.
Or am I getting everything wrong?

Reply:
> returns the same results as `os.Getwd`

It's not `path.Dir(filename)` actually, but `path.Join(path.Dir(filename), "..")`.

> So why do you need `runtime.Caller`?

Given that `regression-core` needs the test to be run with the working directory set to the root of the project, I'm exploiting the path of the current file as a reference to build the path of the root of the project. So I'm using `runtime.Caller` to get the path of `gitbase_back_comp_test.go`.

> And this new comment should belong to `setupWd` function, not `runtime.Caller`.

Yup, it actually belongs to the whole function, gonna move it 👍.